cpp/src/slice2cpp/Gen.cpp

```diff
@@ -908,20 +950,33 @@ Slice::Gen::generate(const UnitPtr& p)
     {
         normalizeMetaData(p, true);
 
+        // Forward declares proxies, classes and structs.
         DeclVisitor declVisitor(H, C, _dllExport);
         p->visit(&declVisitor, false);
 
-        TypesVisitor typesVisitor(H, C, _dllExport);
+        // Generate declarations and definitions for all types except structs, classes, exceptions and proxies.
```

Comment:
Does it generate code for interfaces?

Reply:
No, the interfaces are after, and they are not types (exceptions aren't either).

Reply:
Maybe simpler to list what it does generate code for?